js-intro.js

```diff
@@ -27,70 +43,180 @@ var myDog = "Snoopy"
+console.log(cats)
 
 
+//5b. Write a function that returns only the names of all the non-cats. Expected output: "Itchy" "Daffy" "Ren"
+// Having issues with this one. Having issues accessing object elements. Was eventually able to get it to kind of work with for loops but it would also output an "undefined" at the end.
+console.log("\n5b------------------------")
 
 
+// // Need to do something extra to access object value within class...
+// var nonCats = toonimals.filter(value => value.animal !== "cat")
+// var nonCatsNames = nonCats.name
+// console.log(nonCats.name)
 
 
-// ------------------------------- Consider the following variable:
+// // Define new array maybe to avoid for loops? So I can call out something like newToonimals.first.name. Couldn't get this to work. Trying something else.
+// var newToonimals = [
+//     first: {name: "Itchy", animal: "mouse"},
+//     second: {name: "Stimpy", animal: "cat"},
+//     third: {name: "Daffy", animal: "duck"},
+//     fourth: {name: "Scratchy", animal: "cat"},
+//     fifth: {name: "Ren", animal: "dog"},
+//     sixth: {name: "Felix", animal: "cat"}
+// ]
 
-var toonimals = [ {name: "Itchy", animal: "mouse"}, {name: "Stimpy", animal: "cat"}, {name: "Daffy", animal: "duck"}, {name: "Scratchy", animal: "cat"}, {name: "Ren", animal: "dog"}, {name: "Felix", animal: "cat"}]
 
-// 5a. Write a function that takes in the toonimals array as an argument and returns an array with only the toon objects that are cats. Expected output: [ { name: "Stimpy", animal: "cat" }, { name: "Scratchy", animal: "cat" }, { name: "Felix", animal: "cat" } ]
+// // Trying different methods...ES6 stuff below. Thought it would be able to access the object but seems I'm still missing something.
+// newToonimals = (array) => {
+//     let newArr = array.filter(value => value.animal === "cat")
+//     return newArr
+// }
+// console.log(newToonimals(toonimals))
 
 
+// Going to revert to using for loops. Outputs mostly what I want, but also outputs an additional "undefined".
+function newToonimals(array) {
+    for (let i = 0; i < array.length; i++) {
+        // console.log(toonimals[i])
+        if (array[i].animal !== "cat") {
+            console.log(array[i].name)
+        }
+    }
+}
+console.log(newToonimals(toonimals))
 
 
-//5b. Write a function that returns only the names of all the non-cats. Expected output: "Itchy" "Daffy" "Ren"
+//5c. STRETCH: Create a function that will print a sentence for every toon stating their name and their animal type. Use string interpolation. Expected output: "Itchy is a mouse" "Stimpy is a cat" "Daffy is a duck" "Scratchy is a cat" "Ren is a dog" "Felix is a cat"
+console.log("\n5c------------------------")
 
 
+// // Trying to access object elements within the class, but unable to do so. Going to try it out with for loops.
+// toonType = object => {
+//     `${toonimals.name} is a ${toonimals.animal}`   
+// }
+// console.log(toonType(toonimals))
 
 
```

Comment:
toonType = object => {
  let types = object.map(value => {
    return `${value.name} is a ${value.animal}`
  })
  return types
}
console.log(toonType(toonimals))